CMakeLists.txt

```diff
@@ -1314,65 +1314,63 @@ if (NOT(SKIP_PACKAGING))
 endif()
 
 ################################################################################
-## ERRORS FILE
+## ERRORS FILES
 ################################################################################
 
-if (USE_MAINTAINER_MODE)
-  # If "make clean" removes these files, afterwards neither "make" nor "cmake" work any more.
-  set_directory_properties(PROPERTIES CLEAN_NO_CUSTOM "On")
+# If "make clean" removes these files, afterwards neither "make" nor "cmake" work any more.
+set_directory_properties(PROPERTIES CLEAN_NO_CUSTOM "On")
+
+set(ERROR_FILES
+  lib/Basics/voc-errors.h
+  lib/Basics/error-registry.h
+  js/common/bootstrap/errors.js
+)
 
-  set(ERROR_FILES
-    lib/Basics/voc-errors.h
-    lib/Basics/error-registry.h
-    js/common/bootstrap/errors.js
+set(ERROR_FILES_GEN)
+set(ERRORS_DAT lib/Basics/errors.dat)
+
+foreach (m IN LISTS ERROR_FILES)
+  add_custom_command(
+    OUTPUT ${CMAKE_SOURCE_DIR}/${m}
+    COMMAND ${PYTHON_EXECUTABLE} ./utils/generateErrorfile.py ./${ERRORS_DAT} ./${m}.tmp
```

Comment:
prepend `${CMAKE_SOURCE_DIR}/` pls instead of using `./` 
else LGTM.

Reply:
Note: This is simply the old code without the `if (USE_MAINTAINER_MODE)` around it.

Reply:
Btw the scope of the command is 
```
WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
```

```
  add_custom_command(
    OUTPUT ${CMAKE_SOURCE_DIR}/${m}
    COMMAND ${PYTHON_EXECUTABLE} ./utils/generateErrorfile.py ./${ERRORS_DAT} ./${m}.tmp
    COMMAND ${CMAKE_COMMAND} -E copy_if_different ./${m}.tmp ./${m}
    COMMAND ${CMAKE_COMMAND} -E remove ./${m}.tmp
    DEPENDS ${CMAKE_SOURCE_DIR}/${ERRORS_DAT} ./utils/generateErrorfile.py
    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
    COMMENT  "Building errors files ${m}"
    VERBATIM
  )
```
So using `./` seems to be fine.